runtime/src/main/scala/akka/grpc/internal/ClientState.scala

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.grpc.internal
+
+import java.util.concurrent.atomic.AtomicReference
+
+import akka.Done
+import akka.annotation.InternalApi
+import akka.grpc.GrpcClientSettings
+import akka.stream.{ ActorMaterializer, Materializer }
+import io.grpc.ManagedChannel
+
+import scala.annotation.tailrec
+import scala.concurrent.{ ExecutionContext, Future, Promise }
+import scala.util.Failure
+
+/**
+ * INTERNAL API
+ * Used from generated code so can't be private.
+ *
+ * Client utilities taking care of Channel reconnection and Channel lifecycle in general.
```

Comment:
Is the capital-C "Channel" a reference to something specific?

Reply:
`Channel` is a concept used in gRPC's reference implementation (all languages) and it abstracts the lower layers of the stack (you could have a Channel representing a single socket or a Channel that's a load-balanced pool of connections to many remote endpoints).